parsing/lexer.mll

```diff
@@ -433,25 +433,18 @@ rule token = parse
         lexbuf.lex_curr_p <- { curpos with pos_cnum = curpos.pos_cnum - 1 };
         STAR
       }
-  | ("#" [' ' '\t']* (['0'-'9']+ as num) [' ' '\t']*
-        ("\"" ([^ '\010' '\013' '\"' ] * as name) "\"")?) as directive
-        [^ '\010' '\013'] * newline
-      {
-        match int_of_string num with
-        | exception _ ->
-            (* PR#7165 *)
-            let loc = Location.curr lexbuf in
-            let explanation = "line number out of range" in
-            let error = Invalid_directive (directive, Some explanation) in
-            raise (Error (error, loc))
-        | line_num ->
-           (* Documentation says that the line number should be
-              positive, but we have never guarded against this and it
-              might have useful hackish uses. *)
-            update_loc lexbuf name line_num true 0;
-            token lexbuf
+  | "#"
+      { if lexbuf.lex_start_pos <> 0 then
+          HASH
+        else
+          try directive lexbuf
+          with Failure(_) -> HASH
+      }
+  | newline "#"
+      { update_loc lexbuf None 1 false 1;
+        try directive lexbuf
+        with Failure(_) -> HASH
       }
```

Comment:
This looks reasonable, but is there any chance of unifying these two cases?  Something like
```
    if not (at_beginning_of_line lexbuf.lex_start_p)
    then HASH
    else try directive lexbuf with Failure _ -> HASH
```
where, tentatively,
```
let at_beginning_of_line pos = (pos.pos_cnum = pos.pos_bol)
```
Just a suggestion: I didn't try and I'm not sure it would work.

Reply:
This worked perfectly, thanks.